ffi/CMakeLists.txt

```
@@ -42,37 +32,178 @@ else()
+# GNU g++ and presumably clang on non-apple systems can probably deal with this.
+if(UNIX AND NOT APPLE)
+    set(LLVMLITE_CXX_STATIC_LINK_DEFAULT OFF)
+    option(LLVMLITE_CXX_STATIC_LINK
+           "Enable C++ static linking in llvmlite (GNU Compilers)"
+           ${LLVMLITE_CXX_STATIC_LINK_DEFAULT})
+
+    # See if static libc++ is requested
+    if (LLVMLITE_CXX_STATIC_LINK)
+        check_cxx_compiler_flag(-static-libstdc++ HAVE_STATIC_LIBSTDCXX)
+        if(NOT HAVE_STATIC_LIBSTDCXX)
+            message(FATAL_ERROR "LLVMLITE_CXX_STATIC_LINK was requested but
+                    the compiler does not support the flag")
+        endif()
+        add_definitions(-DLLVMLITE_CXX_STATIC_LINK=1)
+        target_link_options(llvmlite PRIVATE -static-libstdc++)
+    endif()
+endif()
+
+
+# Deal with RTTI, ideally build without it, but allow use if needed.
+if(UNIX)
+    set(LLVMLITE_USE_RTTI_DEFAULT OFF)
+    option(LLVMLITE_USE_RTTI
+        "Build llvmlite with RTTI enabled (Unix systems), default is OFF"
+        ${LLVMLITE_USE_RTTI_DEFAULT})
+
+    set(FLAG_NO_RTTI "-fno-rtti")
+    if(LLVMLITE_USE_RTTI)
+        # strip out -fno-rtti if it's set as the user asked for it to be removed
+        string(REPLACE "${FLAG_NO_RTTI} " "" CMAKE_CXX_FLAGS
+               "${CMAKE_CXX_FLAGS}")
+        message(STATUS "If present -fno-rtti was removed from CXX_FLAGS")
+    else()
+        # check for the flag and add it unconditionally (adding it more than
+        # once is "ok").
+        check_cxx_compiler_flag(${FLAG_NO_RTTI} HAVE_FNO_RTTI)
+        if (NOT HAVE_FNO_RTTI)
+            message(FATAL_ERROR
+            "Compiler must support ${FLAG_NO_RTTI} option if it is requested")
+        endif()
+        target_compile_options(llvmlite PRIVATE ${FLAG_NO_RTTI})
+    endif()
+endif()
+
+
+# Check if the LLVMLITE_SHARED build flag is set. Default is static. This option
+# is baked into the binary for testing purposes.
+# Find the libraries that correspond to the LLVM components needed based on the
+# build flag.
+set(LLVMLITE_SHARED_DEFAULT OFF)
+option(LLVMLITE_SHARED
+       "Enable dynamic linkage against LLVM, default is OFF (i.e. static link)"
+       ${LLVMLITE_SHARED_DEFAULT})
+
+
+if (LLVMLITE_SHARED)
+    check_library_exists(LLVM LLVMGetVersion "" HAVE_LIBRARY_LLVM)
+    if(NOT HAVE_LIBRARY_LLVM)
+        message(FATAL_ERROR "Could not find libLLVM")
+    endif()
     set(llvm_libs LLVM)
+    message(STATUS
+    "LLVMLITE_SHARED is ON, using dynamic linkage against LLVM")
+    add_definitions(-DHAVE_LLVMLITE_SHARED)
+else()
+    # This doesn't work:
+    # llvm_map_components_to_libnames(llvm_libs all)
+    # xref: https://bugs.llvm.org/show_bug.cgi?id=47003
+    # This is a workaround based on knowing what is needed. Do not set llvm_libs
+    # to the cached LLVM_AVAILABLE_LIBS, it may contain the dynamic `LLVM`
+    # library, see llvmlite#1234 (yes, it's #1234, not a placeholder).
```

A tiny nit, but this might be more helpful to the reader: :slightly_smiling_face: 
```suggestion
    # library, see Issue #1234 (https://github.com/numba/llvmlite/issues/1234)
```